src/main/shell-session/shell-env-modifier/shell-env-modifier-registration.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { noop } from "../../../common/utils";
+
+export interface ShellEnvClusterData {
+  context: string;
+  kubeconfigPath: string;
+  id: string;
```

Comment:
would it be enough to just provide the cluster id and let the extension get other things it wants from there?